common/beerocks/bcl/source/beerocks_socket_thread.cpp

```diff
@@ -151,7 +151,7 @@ bool socket_thread::handle_cmdu_message_uds(Socket *sd)
 
     // Check if UDS Header exists
     available_bytes = sd->readBytes(rx_buffer, sizeof(rx_buffer), true, sizeof(message::sUdsHeader),
-                                    true); // PEEK UDS Header, blocking
+                                    true, true); // PEEK UDS Header, blocking, peek the whole header
```

Comment:
`    explicit Socket(SOCKET s, long readTimeout = 1000);
    Socket(SOCKET s, const std::string &peer_ip, int port_port, long readTimeout = 1000);
    Socket(const std::string &uds_path = std::string(), long readTimeout = 1000);
`
All `Socket` contructors set 1s read timeout by default. Probably it applies here as well.

Reply:
This method should not block at all.
It should read all available bytes or up to a complete message, whatever happens first. If there are more bytes available because more than one message has been received, then caller should invoke this method again.